src/main/webapp/app/tasks/task-list/task-list.component.ts

```diff
@@ -12,17 +22,77 @@ import { TaskService } from '../task.service';
   styleUrls: ['./task-list.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TaskListComponent {
+export class TaskListComponent implements OnInit{
 
   @Input() tasks: Task[];
-
+  editMode = false;
+  selectedTasks: Task[] = [];
   @Output() deleted: EventEmitter<Task> = new EventEmitter();
 
-  constructor(@Inject('TaskService') private taskService: TaskService) { }
+  constructor(@Inject('TaskService') private taskService: TaskService, private cdr: ChangeDetectorRef) { }
+
+  ngOnInit(): void {
+    this.taskService.sortTasks();
+  }
 
+  /**
+   * delete a task
+   * @param task
+   */
   delete(task: Task): void {
     this.taskService.delete(task.id).subscribe(() => {
       this.deleted.emit(task);
     });
   }
+
+  /**
+   * change edit mode
+   */
+  editTasks(): void{
+    this.editMode = !this.editMode;
+  }
+
+  /**
+   * check the selected task if true modify css class to be selected
+   * @param task
+   */
+  checkSelectedTask(task: Task): any{
```

Comment:
1. Don't use any as a return type but use proper typings
2. What is the difference to the implementation of `getStatus`
3. avoid the if conditions (in both implementations) and return go for something like `return !!item`